optuna_dashboard/ts/components/GraphHistory.tsx

```diff
@@ -69,6 +71,39 @@ export const GraphHistory: FC<{
       theme.palette.mode,
       markerSize
     )
+    const element = document.getElementById(plotDomId)
+    if (element != null && studies.length >= 1) {
+      // @ts-ignore
+      element.on("plotly_click", function (data) {
+        if (data.points[0].data.mode !== "lines") {
+          let studyId = 1
+          if (data.points[0].data.name.includes("Infeasible Trial of")) {
+            const studyInfo: { id: number; name: string }[] = []
+            studies.forEach((study) => {
+              studyInfo.push({ id: study.id, name: study.name })
+            })
+            const dataPointStudyName = data.points[0].data.name.replace(
+              "Infeasible Trial of ",
+              ""
+            )
+            const targetId = studyInfo.find((s) => s.name === dataPointStudyName)?.id
+            if (targetId !== undefined) {
+              studyId = targetId
+            }
```

Comment:
It did not work for Infeasible trials and has been corrected.
StudyId is identified from the name string of the point, but if you have a better way, please comment.

Reply:
One solution would be to give each plot a little more information, like a Pareto chart.
If this is a good solution, I will create another PR on this and as soon as that it is merged, I will reflect it in this PR.